### PR TITLE
Add Human In the Loop for the BlueNaaS tool

### PR DESCRIPTION
**Pull request by WonderPG**

### CODE REVIEW

File: tests/tools/test_bluenaas_tool.py [hunk 1]
```diff
@@ -17,39 +21,408 @@ async def test_arun(httpx_mock):
...
+                }
+            ],
+            usage_metadata={
+                "input_tokens": 5005,
+                "output_tokens": 287,
+                "total_tokens": 5292,
+            },
+        ),
+    ]
+    # Case where we call bluenaas after validating. Run simu
```
Comment (cszsol) [inline tests/tools/test_bluenaas_tool.py:184]: Small typo

Comment (cszsol) [inline tests/tools/test_bluenaas_tool.py:184]: Usually it is better to split different test cases into separate unit tests to improve readibility. In this case when you run the unit tests you will be able to see easily which case fails and which does not.
If you find yourself writing multiple setup-assert blocks, then often it is a good idea to split them into separate tests.

Another observation is that I don't see a test case for error handling. For example what happens if bluenaas tool is not available ?

**Review by jankrepl — DISMISSED** [review]
I was able to run it locally:)

LGTM:) Thank you!!!

File: tests/tools/test_bluenaas_tool.py (hunk 1 above)
Comment (WonderPG) [reply tests/tools/test_bluenaas_tool.py:182]: I moved the core logic to a function that I test using parametrize. If the bluenaas api is not available, the error handling from langgraph kicks in and we test this one in a separate file

Comment (WonderPG) [reply tests/tools/test_bluenaas_tool.py:182]: Are you referring to the correct line ? I don't see it 😄

**Review by jankrepl — APPROVED** [review]
:) LGTM

**Review by BoBer78 — APPROVED** [review]

**Pull request merged**